tests/lisa/test_energy_model.py

```diff
@@ -0,0 +1,229 @@
+from unittest import TestCase
+
+
+from energy_model import (EnergyModel,
+                          ActiveState, EnergyModelNode, PowerDomain)
+
+little_cluster_active_states = OrderedDict([
+    (1000, ActiveState(power=10)),
+    (2000, ActiveState(power=20)),
+])
+
+little_cluster_idle_states = OrderedDict([
+    ("WFI",              5),
+    ("cpu-sleep-0",      5),
+    ("cluster-sleep-0",  1),
+])
+
+little_cpu_active_states = OrderedDict([
+    (1000, ActiveState(capacity=100, power=100)),
+    (1500, ActiveState(capacity=150, power=150)),
+    (2000, ActiveState(capacity=200, power=200)),
+])
+
+little_cpu_idle_states = OrderedDict([
+    ("WFI",              5),
+    ("cpu-sleep-0",      0),
+    ("cluster-sleep-0",  0),
+])
+
+littles=[0, 1]
+little_pd = PowerDomain(cpus=littles,
+                        idle_states=["cluster-sleep-0"],
+                        parent=None)
+
+def little_cpu_node(cpu):
+    cpu_pd = PowerDomain(cpus=[cpu],
+                         idle_states=["WFI", "cpu-sleep-0"],
+                         parent=little_pd)
+
+    return EnergyModelNode([cpu],
+                           active_states=little_cpu_active_states,
+                           idle_states=little_cpu_idle_states,
+                           power_domain=cpu_pd,
+                           freq_domain=littles)
+
+big_cluster_active_states = OrderedDict([
+    (3000, ActiveState(power=30)),
+    (4000, ActiveState(power=40)),
+])
+
+big_cluster_idle_states = OrderedDict([
+    ("WFI",              8),
+    ("cpu-sleep-0",      8),
+    ("cluster-sleep-0",  2),
+])
+
+big_cpu_active_states = OrderedDict([
+    (3000, ActiveState(capacity=300, power=300)),
+    (4000, ActiveState(capacity=400, power=400)),
+])
+
+big_cpu_idle_states = OrderedDict([
+    ("WFI",              9),
+    ("cpu-sleep-0",      0),
+    ("cluster-sleep-0",  0),
+])
+
+bigs=[2, 3]
+big_pd = PowerDomain(cpus=bigs,
+                     idle_states=["cluster-sleep-0"],
+                     parent=None)
+
+def big_cpu_node(cpu):
+    cpu_pd = PowerDomain(cpus=[cpu],
+                         idle_states=["WFI", "cpu-sleep-0"],
+                         parent=big_pd)
+
+    return EnergyModelNode([cpu],
+                           active_states=big_cpu_active_states,
+                           idle_states=big_cpu_idle_states,
+                           power_domain=cpu_pd,
+                           freq_domain=bigs)
+
+levels = {
+    "cluster": [
+        EnergyModelNode(cpus=bigs,
+                        active_states=big_cluster_active_states,
+                        idle_states=big_cluster_idle_states),
+        EnergyModelNode(cpus=littles,
+                        active_states=little_cluster_active_states,
+                        idle_states=little_cluster_idle_states)
+    ],
+    "cpu": [little_cpu_node(0), little_cpu_node(1),
+            big_cpu_node(2), big_cpu_node(3)]
+}
+
+em = EnergyModel(levels=levels)
+
+@unittest.skip("No worky")
```

Comment:
Yep, this should not be here! I didn't actually mean to commit the unit tests, they aren't really ready.